app/ui.js

```diff
@@ -1099,14 +1096,49 @@ var UI;
             // Don't display the connection settings until we're actually disconnected
         },
 
+        reconnect: function() {
+            delete UI.reconnect_callback;
+
+            // if reconnect has been disabled in the meantime, do nothing.
+            if (UI.inhibit_reconnect) {
+                return;
+            }
+
+            UI.connect();
+        },
+
         disconnectFinished: function (rfb, reason) {
+            var reconnect = UI.getSetting('reconnect');
+
+            if ((reconnect === 'true' || reconnect == '1') && !UI.inhibit_reconnect) {
```

Comment:
Well this is embarrassing, that's my code and it's fairly recent. :)

But I had a close look, and the magic starts when you start using `UI.getSetting()` rather than `WebUtil.getConfigVar()`. The former checks if the setting is attached to a check box, and if so tries to convert things to a boolean.